src/main/java/io/personium/common/auth/token/AbstractLocalToken.java

```diff
@@ -192,29 +206,29 @@ final String doCreateTokenString(final String[] extendedFields) {
     }
 
     /**
-     * 指定のIssuer向けのIV (Initial Vector)を生成して返します.
-     * IVとしてissuerの最後の最後の１６文字を逆転させた文字列を用います。
-     * これにより、違うIssuerを想定してパースすると、パースに失敗する。
+     * Generate an IV (Initial Vector) and return it for a specified token issuer.
      * @param issuer Issuer URL
      * @return Initial Vector Byte array
      */
     protected static byte[] getIvBytes(final String issuer) {
         try {
-            return StringUtils.reverse("123456789abcdefg" + issuer)
-                    .substring(0, IV_BYTE_LENGTH).getBytes(CharEncoding.UTF_8);
+            MessageDigest md = MessageDigest.getInstance(SHA256);
+            byte[] hash = md.digest(issuer.getBytes(CharEncoding.UTF_8));
+            return Arrays.copyOfRange(hash, IV_OFFSET_FROM_SHA256, IV_OFFSET_FROM_SHA256 + IV_BYTE_LENGTH);
         } catch (UnsupportedEncodingException e) {
             throw new RuntimeException(e);
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
```

Comment:
Catching multiple exception is better.
https://docs.oracle.com/javase/7/docs/technotes/guides/language/catch-multiple.html

```java
        } catch (UnsupportedEncodingException|NoSuchAlgorithmException  e) {
            throw new RuntimeException(e);
        } 
```

Reply:
Thanks